### ISSUE TITLE
Section headings counted as list elements by screen readers

### ISSUE THREAD

**Issue by wkillerud** [timc1/kbar#269]
Hey, and thanks for building and sharing this lovely tool ❤️ 

I did some testing with VoiceOver, and noticed an issue with how list items are counted.

## Steps to reproduce

1. Visit https://kbar.vercel.app
2. Start VoiceOver
3. Open the kbar with Cmd + K. If using Safari, type any key and delete the character you entered [^1].
5. Press Arrow down.

Notice how VoiceOver reads Home as _2 of 10_, and Docs as  _3 of 10_. Also notice how it's impossible to reach _1 of 10_.

## Debugging

- [Everything in the `listbox` is an `option`](https://github.com/timc1/kbar/blob/e1faecd6b71edd1c1b1edc8c9b5faf6a28805dcd/src/KBarResults.tsx#L160), including section headings, so everything is counted as an item by VoiceOver.
- With `role="presentation"` set for section headings they are still counted.
- Setting `aria-hidden="true"` for section headings solves the counting problem, but obviously makes the heading inaccessible. A potential workaround is to set an `aria-label` that includes the section heading, but that quickly gets verbose.

I ended up setting `aria-hidden="true"` on the section headings with a local patch for now. Of course the ideal solution would be to keep the headings accessible _and_ count the options correctly. I just couldn't figure out how 😅 

## Versions

- `kbar` versions `0.1.0-beta.38` and `0.1.0-beta.39`
- macOS 13
- Safari 16.1
- Chrome 108
- Firefox 107

---

[^1]: Haven't been able to track down why, but for some reason VoiceOver in Safari doesn't announce the active list item on first render. Works fine in Firefox and Chrome. Also works fine in Safari after inputing a single character. 🤷
